### PR TITLE
Renamed module mxcube3 to mxcubeweb and updated all references to mxcube3

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
Draft because it needs some more checking  :)

### CODE REVIEW

File: debian-install.sh [hunk 1]
```diff
@@ -7,11 +7,11 @@ system_has() {
 }
 
 mxcube_source() {
-  echo "https://github.com/mxcube/mxcube3.git"
+  echo "https://github.com/mxcube/mxcubewen.git"
```
Comment (fabcor-maxiv) [inline debian-install.sh]: Typo

File: mxcubeweb/__version__.py
Comment (fabcor-maxiv) [inline mxcubeweb/__version__.py:1]: _[Side note: this whole file is bad practice (in my opinion), do we truly need this file at all? I might address this in separate ticket and PR.]_

File: mxcubeweb/core/components/beamline.py [hunk 2]
```diff
@@ -242,16 +241,6 @@ def beamline_abort_action(self, name):
...
-        try:
-            cmds = HWR.beamline.beamline_actions.get_commands()
-        except Exception:
-            cmds = []
-
-        for cmd in cmds:
-            if cmd.name() == name:
-                cmd.abort()
-
-        # Annotated command
```
Comment (fabcor-maxiv) [inline mxcubeweb/core/components/beamline.py]: Unrelated change? Can this be done outside of this pull request?

File: mxcubeweb/core/components/beamline.py [hunk 3]
```diff
@@ -268,25 +257,6 @@ def beamline_run_action(self, name, params):
...
-                    logging.getLogger("user_level_log").info(
-                        "Starting %s(%s)",
-                        cmd.name(),
-                        ", ".join(map(str, params)),
-                    )
-                    cmd(*params)
-                except Exception as ex:
-                    err = str(sys.exc_info()[1])
-                    raise Exception(str(err)) from ex
-        # Annotated command
```
Comment (fabcor-maxiv) [inline mxcubeweb/core/components/beamline.py]: Unrelated change? Can this be done outside of this pull request?

File: mxcubeweb/core/components/lims.py.orig
Comment (fabcor-maxiv) [inline mxcubeweb/core/components/lims.py.orig:1]: What is this file?

File: mxcubeweb/core/components/queue.py.orig
Comment (fabcor-maxiv) [inline mxcubeweb/core/components/queue.py.orig:1]: What is this file?

File: mxcubeweb/core/models/generic.py.orig
Comment (fabcor-maxiv) [inline mxcubeweb/core/models/generic.py.orig:1]: Useless file.

File: mxcubeweb/routes/login.py.orig
Comment (fabcor-maxiv) [inline mxcubeweb/routes/login.py.orig:1]: Useless file?

File: mxcubeweb/routes/main.py.orig
Comment (fabcor-maxiv) [inline mxcubeweb/routes/main.py.orig:1]: Useless file?

File: mxcubeweb/server.py.orig
Comment (fabcor-maxiv) [inline mxcubeweb/server.py.orig:1]: Useless file?

File: pyproject.toml [hunk 4]
```diff
@@ -20,7 +20,7 @@ classifiers = [
     "Natural Language :: English",
     "Topic :: Scientific/Engineering",
 ]
-packages = [{include = "mxcube3", from = "."}]
+packages = [{include = "mxcubeweb", from = "."}]
```
Comment (fabcor-maxiv) [inline pyproject.toml]: This line is probably useless now that the distribution package name matches the name of the top-level import package.

File: ui/src/containers/BeamlineSetupContainer.jsx [hunk 5]
```diff
@@ -296,13 +296,13 @@ class BeamlineSetupContainer extends React.Component {
...
               <span className="blstatus-item">
-                {this.props.beamline.hardwareObjects.machine_info && (
+                {this.props.beamline.hardwareObjects.machine_info ? (
                   <MachInfo
                     info={
                       this.props.beamline.hardwareObjects.machine_info.value
                     }
                   />
-                )}
+                ) : null}
```
Comment (fabcor-maxiv) [inline ui/src/containers/BeamlineSetupContainer.jsx]: Unrelated change? I prefer this change would not be introduced in this pull request.

File: CONTRIBUTING.md
Comment (fabcor-maxiv) [inline CONTRIBUTING.md:1]: Too bad this introduces unrelated changes. It makes it hard to see if the renaming was done correctly. Is it possible to do unrelated changes in different pull requests?

File: docs/Makefile [hunk 6]
```diff
@@ -87,9 +87,9 @@ qthelp:
 	@echo
 	@echo "Build finished; now you can run "qcollectiongenerator" with the" \
 	      ".qhcp project file in $(BUILDDIR)/qthelp, like this:"
-	@echo "# qcollectiongenerator $(BUILDDIR)/qthelp/MXCuBE3.qhcp"
+	@echo "# qcollectiongenerator $(BUILDDIR)/qthelp/MXCuBEWeb.qhcp"
 	@echo "To view the help file:"
-	@echo "# assistant -collectionFile $(BUILDDIR)/qthelp/MXCuBE3.qhc"
+	@echo "# assistant -collectionFile $(BUILDDIR)/qthelp/MXCuBEWeb.qhc"
```
Comment (fabcor-maxiv) [inline docs/Makefile:92]: _[Side note: we should probably clean up the whole Makefile, it is full of things we do not use or need. For example this change here, I am not sure it would still work, no idea because I have never ever used this part of Sphinx Makefile. Maybe I will take the time to clean this up next time I work on the Sphinx things, I probably should have done it on the first round of Sphinx doc reboot.]_

File: docs/Makefile [hunk 7]
```diff
@@ -104,8 +104,8 @@ devhelp:
 	@echo
 	@echo "Build finished."
 	@echo "To view the help file:"
-	@echo "# mkdir -p $$HOME/.local/share/devhelp/MXCuBE3"
-	@echo "# ln -s $(BUILDDIR)/devhelp $$HOME/.local/share/devhelp/MXCuBE3"
+	@echo "# mkdir -p $$HOME/.local/share/devhelp/MXCuBEWeb"
+	@echo "# ln -s $(BUILDDIR)/devhelp $$HOME/.local/share/devhelp/MXCuBEWeb"
```
Comment (fabcor-maxiv) [inline docs/Makefile:108]: _[Same side note as above.]_

File: mxcube3/video/__init__.py
Comment (fabcor-maxiv) [inline mxcube3/video/__init__.py:1]: Is this whole directory removed? Is it on purpose? Why?

I can not exclude that it is the GitHub UI playing tricks on me...

File: mxcubeweb/.editorconfig
Comment (fabcor-maxiv) [inline mxcubeweb/.editorconfig:1]: Change unrelated to scope of pull request.

File: mxcubeweb/.vscode/settings.json
Comment (fabcor-maxiv) [inline mxcubeweb/.vscode/settings.json:1]: Change unrelated to scope of pull request.

File: CONTRIBUTING.md
Comment (fabcor-maxiv) [inline CONTRIBUTING.md:1]: Too bad this file contains a lot of unrelated changes. It makes it hard to check if the renaming was done correctly.

File: mxcubeweb/__version__.py
Comment (fabcor-maxiv) [inline mxcubeweb/__version__.py:1]: _[Side note: this whole file is bad practice. Do we really need this file? I will likely address this in separate issue and PR.]_

**Review by fabcor-maxiv — CHANGES_REQUESTED** [review]

File: mxcubeweb/__version__.py
Comment (marcus-oscarsson) [reply mxcubeweb/__version__.py:1]: Hm what do you mean ? :)

File: CONTRIBUTING.md
Comment (marcus-oscarsson) [reply CONTRIBUTING.md:1]: Hm, yes its pre-commit that required the formatting changes. Altough the name change is quite easy to see IMO

File: mxcube3/video/__init__.py
Comment (marcus-oscarsson) [reply mxcube3/video/__init__.py:1]: No, it was removed its not used any longer. So its not GitHub UI :)

File: mxcubeweb/core/components/beamline.py (hunk 2 above)
Comment (marcus-oscarsson) [reply mxcubeweb/core/components/beamline.py]: Ah, yes Ill extract these changes and make another PR

File: debian-install.sh (hunk 1 above)
Comment (marcus-oscarsson) [reply debian-install.sh]: Wopsi

**Comment by marcus-oscarsson** [comment]
@fabcor-maxiv thanks for having a look. 

I removed the files and the code in `beamline.py` that was committed by mistake. 

The `video` module is no longer used in `mxcubeweb`. 

For the `__version__` file it would be interesting to know why you think its bad practice. I have no real opinion myself, its one way I know of to get the version information. We do use the version string and send it to the client, this is simply one way of doing that. Its not really used by the client so we could eventually remove it for that reason. However I think its quite handy to be able to get which version one is importing ?

File: docs/Makefile (hunk 6 above)
Comment (marcus-oscarsson) [reply docs/Makefile:92]: Probably a good idea

File: mxcubeweb/__version__.py
Comment (fabcor-maxiv) [reply mxcubeweb/__version__.py:1]: I guess it's just one of my pet peeves. When I have time I will investigate if we can improve without losing functionality :)

File: docs/Makefile (hunk 6 above)
Comment (fabcor-maxiv) [reply docs/Makefile:92]: I will put it on my (mental) todo list :)

File: mxcubeweb/core/components/beamline.py (hunk 2 above)
Comment (fabcor-maxiv) [reply mxcubeweb/core/components/beamline.py]: Thanks :)

File: mxcubeweb/core/components/beamline.py (hunk 3 above)
Comment (fabcor-maxiv) [reply mxcubeweb/core/components/beamline.py]: Thanks :)

File: mxcubeweb/.editorconfig
Comment (fabcor-maxiv) [reply mxcubeweb/.editorconfig:1]: Thanks :)

File: mxcubeweb/.vscode/settings.json
Comment (fabcor-maxiv) [reply mxcubeweb/.vscode/settings.json:1]: Thanks :)

File: ui/src/containers/BeamlineSetupContainer.jsx (hunk 5 above)
Comment (fabcor-maxiv) [reply ui/src/containers/BeamlineSetupContainer.jsx]: @marcus-oscarsson Was this a change required by the linter?

File: pyproject.toml (hunk 4 above)
Comment (marcus-oscarsson) [reply pyproject.toml]: Oh yes

File: ui/src/containers/BeamlineSetupContainer.jsx (hunk 5 above)
Comment (marcus-oscarsson) [reply ui/src/containers/BeamlineSetupContainer.jsx]: No, not required. just nicer. Actually it should not even be there, ill remove it.

Comment (marcus-oscarsson) [reply ui/src/containers/BeamlineSetupContainer.jsx]: Hm, actually its not nicer, just a mistake :)

**Comment by fabcor-maxiv** [comment]
Thanks for addressing my comments. Seems like it is good for me now :)


> I removed the files and the code in `beamline.py` that was committed by mistake.

Thanks


> The `video` module is no longer used in `mxcubeweb`.

OK


> For the `__version__` file it would be interesting to know why you think its bad practice. I have no real opinion myself, its one way I know of to get the version information. We do use the version string and send it to the client, this is simply one way of doing that. Its not really used by the client so we could eventually remove it for that reason. However I think its quite handy to be able to get which version one is importing ?

I guess it's just another one of my (many?) pet peeves. I commented on this because at first GitHub UI made it look like it was a newly added file, but it was just renamed. When I have time I will investigate if we can improve this without losing functionality. :)

File: pyproject.toml (hunk 4 above)
Comment (fabcor-maxiv) [reply pyproject.toml]: I see it's removed now, thanks :)

File: ui/src/containers/BeamlineSetupContainer.jsx (hunk 5 above)
Comment (fabcor-maxiv) [reply ui/src/containers/BeamlineSetupContainer.jsx]: I see :D
